src/mongoChangeStream.ts

```diff
@@ -283,15 +283,21 @@ export function initSync<ExtendedEvents extends EventEmitter.ValidEventTypes>(
       state.change('started')
 
       const nextChecker = safelyCheckNext(cursor)
-      let doc: Document | null
-      // Process documents
-      while ((doc = await nextChecker.getNext())) {
-        debug('Initial scan doc %O', doc)
-        await queue.enqueue(toChangeStreamInsert(doc))
-        await pause.maybeBlock()
+
+      try {
+        let doc: Document | null
+        // Process documents
+        while ((doc = await nextChecker.getNext())) {
+          debug('Initial scan doc %O', doc)
+          await queue.enqueue(toChangeStreamInsert(doc))
+          await pause.maybeBlock()
+        }
+        // Flush the queue
+        await queue.flush()
+      } catch (e) {
+        emit('error', { error: e, initialScan: true })
```

Comment:
Open question: should we avoid setting the initialScanCompletedOn key and emitting initialScanComplete here? The way I have it now, if there's an error, we proceed to the code below and will do those things, even if there was an error, which is a change from current behavior.